newIDE/app/src/UI/SortableVirtualizedItemList/index.js

```diff
@@ -98,75 +133,55 @@ export default class SortableVirtualizedItemList<Item> extends React.Component<
-                              <ItemRow
-                                item={item}
-                                itemName={itemName}
-                                isBold={isItemBold ? isItemBold(item) : false}
-                                onRename={newName =>
-                                  this.props.onRename(item, newName)
-                                }
-                                editingName={nameBeingEdited}
-                                getThumbnail={
-                                  getItemThumbnail
-                                    ? () => getItemThumbnail(item)
-                                    : undefined
-                                }
-                                selected={selectedItems.indexOf(item) !== -1}
-                                onItemSelected={this.props.onItemSelected}
-                                errorStatus={
-                                  erroredItems
-                                    ? erroredItems[itemName] || ''
-                                    : ''
-                                }
-                                buildMenuTemplate={() =>
-                                  this.props.buildMenuTemplate(item, index)
-                                }
-                                onEdit={onEditItem}
-                                hideMenuButton={windowWidth === 'small'}
-                                connectIconDragSource={
-                                  // If on a touch screen, only set the icon to be draggable.
+                      {nameBeingEdited ? (
+                        this._renderItemRow(item, index, windowWidth)
+                      ) : (
+                        <DragSourceAndDropTarget
+                          beginDrag={() => {
+                            this.props.onItemSelected(item);
+                            return {};
+                          }}
+                          canDrop={() =>
+                            canMoveSelectionToItem
+                              ? canMoveSelectionToItem(item)
+                              : true
+                          }
+                          drop={() => {
+                            onMoveSelectionToItem(item);
+                          }}
+                        >
+                          {({
+                            connectDragSource,
+                            connectDropTarget,
+                            isOver,
+                            canDrop,
+                          }) => {
+                            // Add an extra div because connectDropTarget/connectDragSource can
+                            // only be used on native elements
+                            const dropTarget = connectDropTarget(
+                              <div>
+                                {isOver && <DropIndicator canDrop={canDrop} />}
+                                {// If on a touch screen, only set the icon to be draggable.
```

Comment:
This comment is now out of place, should be relocated next to what it describes.

Reply:
I'll update it :)